### PR TITLE
ENH: Raise a dedicated JobError if any subjob fails (in --follow mode only)

### LINKED ISSUE #512: run and jobs --fetch: fail with non-0 exit codes if anything (subjob, fetch, etc) fails

### PR DESCRIPTION
**Pull request by yarikoptic**
Closes #512

### CODE REVIEW

File: reproman/support/exceptions.py [hunk 1]
```diff
@@ -197,3 +197,22 @@ class SSHAccessDeniedViaAuthKeys(BaseException):
     """
     def __init__(self, user):
         self.msg = "access for user '%s' denied via authorized_keys" % user
+
+
+#
+# Run related exceptions
+#
+class JobError(RuntimeError):
+    """reproman run  execution related error.
```
Comment (kyleam) [inline reproman/support/exceptions.py:206]: nit-pick: extra space

File: reproman/support/exceptions.py [hunk 2]
```diff
@@ -197,3 +197,22 @@ class SSHAccessDeniedViaAuthKeys(BaseException):
...
+    """
+    def __init__(self, *args, failed=None):
+        super().__init__(*args)
```
Comment (kyleam) [inline reproman/support/exceptions.py:209]: Why match/propagate `RuntimeError`s `args` but not its `kwargs`?

File: reproman/support/exceptions.py [hunk 3]
```diff
@@ -197,3 +197,22 @@ class SSHAccessDeniedViaAuthKeys(BaseException):
     """
     def __init__(self, user):
         self.msg = "access for user '%s' denied via authorized_keys" % user
+
+
+#
+# Run related exceptions
+#
+class JobError(RuntimeError):
```
Comment (kyleam) [inline reproman/support/exceptions.py:205]: I think it'd make sense to position this next the `OrchestratorError`.

File: reproman/support/exceptions.py (hunk 1 above)
Comment (yarikoptic) [reply reproman/support/exceptions.py:206]: argh - my habit for visual separation of commands from the text. Will do as """`reproman run`"""

File: reproman/support/exceptions.py (hunk 3 above)
Comment (yarikoptic) [reply reproman/support/exceptions.py:205]: true, will do

File: reproman/support/exceptions.py (hunk 2 above)
Comment (yarikoptic) [reply reproman/support/exceptions.py:209]: it seems that it takes no kwargs:
```
In [3]: RuntimeError(buga="duga")                                                                                
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-825a24e4a3fd> in <module>
----> 1 RuntimeError(buga="duga")

TypeError: RuntimeError() takes no keyword arguments
```
so I will keep as is

Comment (kyleam) [reply reproman/support/exceptions.py]: Thank you for checking.  When inspecting objects with `pydoc`, I should remember to check for the second line below and not trust the signature if I see it, though in this case I don't think its suggestion is useful.

```
 |  __init__(self, /, *args, **kwargs)
 |      Initialize self.  See help(type(self)) for accurate signature.
```

**Pull request merged**